File: C++ Programming/Competitive Programming/Coding Challenges 11/program5.cpp
// 5. Write a program which returns difference between Even factorial and odd factorial of given number.
// Input : 5
// Output : -7 (8 - 15)
// Input : -5
// Output : -7 (8 - 15)
// Input : 10
// Output : 2895 (3840 - 945)

#include <iostream>
using namespace std;

int FactorialDiff(int iNo)
{
    int iEvenFact = 1;
    int iOddFact = 1;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    for (int i = 2; i <= iNo; i += 2)
    {
        iEvenFact = iEvenFact * i;
    }

    for (int i = 1; i <= iNo; i += 2)
    {
        iOddFact = iOddFact * i;
    }

    return iEvenFact - iOddFact;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number: ";
    cin >> iValue;

    iRet = FactorialDiff(iValue);

    cout << "Factorial difference is " << iRet;

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 15/program5.java
/*
5.Write a program which accept number from user and return difference between summation of even digits and summation of odd digits.
Input : 2395
Output : -15 (2 - 17)
Input : 1018
Output : 6 (8 - 2)
Input : 8440
Output : 16 (16 - 0)
Input : 5733
Output : -18 (0 - 18)
*/

import java.util.Scanner;

class program5
{
    static int CountDiff(int iNo)
    {
        int iEvenSum = 0;
        int iOddSum = 0;

        String str = Integer.toString(iNo);

        for (int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            if (ch >= '0' && ch <= '9')
            {
                int iDigit = ch - '0';

                if (iDigit % 2 == 0)
                {
                    iEvenSum += iDigit;
                }
                else
                {
                    iOddSum += iDigit;
                }
            }
        }

        return iEvenSum - iOddSum;
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter Number : ");
        iValue = sobj.nextInt();

        iRet = CountDiff(iValue);

        System.out.print(iRet);

        sobj.close();
    }
}